TagsterWebAPI/libraries/Tagster.Auth/Factory/TokenValidationFactory.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Microsoft.IdentityModel.Tokens;
+using Tagster.Auth.Options;
+
+namespace Tagster.Auth.Factory
+{
+    internal static class TokenValidationFactory
+    {
+        public static TokenValidationParameters CreateParameters(JwtOptions options)
+        {
+            return new()
+            {
+                RequireAudience = options.RequireAudience,
+                ValidIssuer = options.ValidIssuer,
+                ValidIssuers = options.ValidIssuers,
+                ValidateActor = options.ValidateActor,
+                ValidAudience = options.ValidAudience,
+                ValidAudiences = options.ValidAudiences,
+                ValidateAudience = options.ValidateAudience,
+                ValidateIssuer = options.ValidateIssuer,
+                ValidateLifetime = options.ValidateLifetime,
+                ValidateTokenReplay = options.ValidateTokenReplay,
+                ValidateIssuerSigningKey = options.ValidateIssuerSigningKey,
+                SaveSigninToken = options.SaveSignInToken,
+                RequireExpirationTime = options.RequireExpirationTime,
+                RequireSignedTokens = options.RequireSignedTokens,
+                ClockSkew = TimeSpan.Zero
+            };
+        }
+
+        public static void AddAuthenticationType(this TokenValidationParameters tokenValidationParameters,
+            JwtOptions options)
+        {
+            if (!string.IsNullOrWhiteSpace(options.AuthenticationType))
+            {
+                tokenValidationParameters.AuthenticationType = options.AuthenticationType;
+            }
+        }
+
+        public static void AddIssuerSigningKey(this TokenValidationParameters tokenValidationParameters,
+            JwtOptions options)
+        {
+            var hasCertificate = false;
+            if (options.Certificate is { })
+            {
+                X509Certificate2 certificate = null;
+                var password = options.Certificate.Password;
+                var hasPassword = !string.IsNullOrWhiteSpace(password);
+                if (!string.IsNullOrWhiteSpace(options.Certificate.Location))
+                {
+                    certificate = hasPassword
+                        ? new X509Certificate2(options.Certificate.Location, password)
+                        : new X509Certificate2(options.Certificate.Location);
+                    var keyType = certificate.HasPrivateKey ? "with private key" : "with public key only";
+                    Console.WriteLine($"Loaded X.509 certificate from location: '{options.Certificate.Location}' {keyType}.");
```

Comment:
Here and in other places I'd use some kind of a configurable log provider. If yuo don't want to do it in the scope of this task, create a new one